src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs

```diff
@@ -432,7 +432,31 @@ private KeyedByTypeCollection<IOperationBehavior> GetIOperationBehaviorAttribute
                 // the "other" interface (not the one implemented by implType)
                 if (ifaceMethod != null)
                 {
-                    MethodInfo implMethod = GetCorrespondingMethodFromType(implType, opMethod);
+                    MethodInfo implMethod = null;
+                    try
+                    {
+                        GetCorrespondingMethodFromType(implType, opMethod);
+                    }
+
+                    catch (TypeAccessException)
+                    {
+#if FEATURE_NETNATIVE
```

Comment:
Having a catch/rethrow for .Net Core because of some .NET Native code doesn't feel right. Try/Catch blocks reduce the ability for the JIT to do some optimizations too so I would rather only have the try/catch for .NET Native